pkg/storage/replica_command.go

```diff
@@ -3277,14 +3277,16 @@ func (r *Replica) ChangeReplicas(
 	return nil
 }
 
+// sendSnapshot sends a snapshot of the replica state to the specified
+// replica. This is used for both preemptive snapshots that are performed
+// before adding a replica to a range, and for Raft-initiated snapshots that
+// are used to bring a replica up to date that has fallen too far
+// behind. Currently only invoked from replicateQueue and raftSnapshotQueue. Be
+// careful about adding additional calls as applying a snapshot is moderately
+// expensive. Too many concurrent snapshots can overwhelm the remote node.
```

Comment:
This concurrency affects the local store rather than the remote node, which has its own throttling via reservations, no?